tracer/src/Datadog.Trace/Debugger/DynamicInstrumentation.cs

```diff
@@ -87,87 +82,76 @@ internal DynamicInstrumentation(
 
         public bool IsDisposed => Volatile.Read(ref _disposeState) != 0;
 
-        public bool IsInitialized => Volatile.Read(ref _initState) == 2;
+        public bool IsInitialized { get; private set; }
 
         internal void Initialize()
         {
-            var originalState = Interlocked.CompareExchange(ref _initState, 1, 0);
-
-            // If we weren't in "not initialized" state, return early
-            if (originalState != 0)
+            if (!_settings.DynamicInstrumentationEnabled)
             {
+                Log.Information("Dynamic Instrumentation is disabled. To enable it, please set DD_DYNAMIC_INSTRUMENTATION_ENABLED environment variable to 'true'.");
                 return;
             }
 
-            try
-            {
-                if (!_settings.DynamicInstrumentationEnabled)
-                {
-                    Log.Information("Dynamic Instrumentation is disabled. To enable it, please set DD_DYNAMIC_INSTRUMENTATION_ENABLED environment variable to 'true'.");
-                    // Reset to "not initialized"
-                    Interlocked.Exchange(ref _initState, 0);
-                    return;
-                }
-
-                Log.Information("Dynamic Instrumentation initialization started");
+            Log.Information("Dynamic Instrumentation initialization started");
```

Comment:
nit: this should probably be debug

Reply:
Fixed in 9cc8537e8fce4bd242b9c5798bfc51c00d140e38